### ISSUE TITLE
Http2 Protocol Error, errorCode = ProtocolError

### ISSUE THREAD

**Issue by JamesNK** [dotnet/corefx#36463]
When I make HTTP/2 requests this error is thrown internally every time:

```
System.Net.Http.Http2Connection.Http2ProtocolException: 'Http2 Protocol Error, errorCode = ProtocolError'
   at System.Net.Http.Http2Connection.Http2Stream.OnResponseHeader(ReadOnlySpan`1 name, ReadOnlySpan`1 value)
```

![image](https://user-images.githubusercontent.com/303201/55192383-32160d00-5209-11e9-9fb5-d466b2095006.png)

**Comment by davidsh** [comment]
@rmkerr

**Comment by rmkerr** [comment]
Thanks for the report @JamesNK. Can you share any more details about the repro?

**Comment by JamesNK** [comment]
Wireshark log

[httpclient-grpc.zip](https://github.com/dotnet/corefx/files/3020212/httpclient-grpc.zip)

**Comment by JamesNK** [comment]
From the stacktrace it looks like it is thrown here:

https://github.com/dotnet/corefx/blob/304c0194b16a1f26c14cda92e2532efcf126c99f/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs#L118-L127

**Comment by rmkerr** [comment]
This is actually simpler than I initially thought. These gRPC responses include trailing headers, which aren't supported yet. Support is introduced in pr #36128.

**Comment by JamesNK** [comment]
Ok, so is it best to wait until that PR lands and then retest?

**Comment by rmkerr** [comment]
Yep. If you're interested in getting some tests in ASAP you can build the branch with trailing header support and test against those binaries. Otherwise it will be easiest for the change to just make it into master.

**Comment by wfurt** [comment]
BTW older version of SockertHttpClient ignore trailing headers. You can try preview3 build instead of latest @JamesNK

**Comment by wfurt** [comment]
Should be fixed by  #36128. Let me know if you have any other troubles @JamesNK
